packages/office-ui-fabric-react/src/components/Dialog/examples/Dialog.Basic.Example.tsx

```diff
@@ -1,81 +1,57 @@
+};
+
+export const DialogBasicExample: React.FunctionComponent = () => {
+  const [hideDialog, { toggle: toggleHideDialog }] = useBoolean(true);
+  const [isDraggable, { toggle: toggleIsDraggable }] = useBoolean(false);
+  const labelId: string = useId('dialogLabel');
+  const subTextId: string = useId('subTextLabel');
+  const dragOptions = {
     moveMenuItemText: 'Move',
     closeMenuItemText: 'Close',
     menu: ContextualMenu,
   };
-
-  public render() {
-    const { hideDialog, isDraggable } = this.state;
-    return (
-      <div>
-        <Checkbox label="Is draggable" onChange={this._toggleDraggable} checked={isDraggable} />
-        <DefaultButton secondaryText="Opens the Sample Dialog" onClick={this._showDialog} text="Open Dialog" />
-        <label id={this._labelId} className={screenReaderOnly}>
-          My sample Label
-        </label>
-        <label id={this._subTextId} className={screenReaderOnly}>
-          My Sample description
-        </label>
-
-        <Dialog
-          hidden={hideDialog}
-          onDismiss={this._closeDialog}
-          dialogContentProps={{
-            type: DialogType.normal,
-            title: 'Missing Subject',
-            closeButtonAriaLabel: 'Close',
-            subText: 'Do you want to send this message without a subject?',
-          }}
-          modalProps={{
-            titleAriaId: this._labelId,
-            subtitleAriaId: this._subTextId,
-            isBlocking: false,
-            styles: { main: { maxWidth: 450 } },
-            dragOptions: isDraggable ? this._dragOptions : undefined,
-          }}
-        >
-          <DialogFooter>
-            <PrimaryButton onClick={this._closeDialog} text="Send" />
-            <DefaultButton onClick={this._closeDialog} text="Don't send" />
-          </DialogFooter>
-        </Dialog>
-      </div>
-    );
-  }
-
-  private _showDialog = (): void => {
-    this.setState({ hideDialog: false });
-  };
-
-  private _closeDialog = (): void => {
-    this.setState({ hideDialog: true });
-  };
-
-  private _toggleDraggable = (): void => {
-    this.setState({ isDraggable: !this.state.isDraggable });
-  };
-}
+  return (
+    <>
+      <Checkbox label="Is draggable" onChange={toggleIsDraggable} checked={isDraggable} />
+      <DefaultButton secondaryText="Opens the Sample Dialog" onClick={toggleHideDialog} text="Open Dialog" />
+      <label id={labelId} className={screenReaderOnly}>
+        My sample Label
+      </label>
+      <label id={subTextId} className={screenReaderOnly}>
+        My Sample description
+      </label>
+
+      <Dialog
+        hidden={hideDialog}
+        onDismiss={toggleHideDialog}
+        dialogContentProps={dialogContentProps}
+        modalProps={{
+          titleAriaId: labelId,
+          subtitleAriaId: subTextId,
+          isBlocking: false,
+          styles: { main: { maxWidth: 450 } },
+          dragOptions: isDraggable ? dragOptions : undefined,
+        }}
```

Comment:
You should consider using `useMemo` for this object and moving it as a const before the return statement.